### DISCUSSION TITLE
Raspberry Pico and SSD1306 with Arduino

### DISCUSSION THREAD

**Discussion by juergen-erstling** [olikraus/u8g2#1889] [General]
Hi all

I try to use the u8g2 software on an Raspberry Pico. Does anyone have experience with connecting an OLED display to the Raspberry Pico or a sample program?
I can use the library on an Arduino Uno or Mega without any problems. I can compile the same program for the pico, but nothing is displayed.
The pico has two wire channel  I2C0 and I2C1. I tried I2C0 with the default ports GP5 (SCL) and GP4 (SDA).

**Comment by pereltje** [comment]
Hello Juergen,
on a pico W this works for me:
SDA of the display wired to GP4
SCL of the display wired to GP5
and this display constructor:
U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R2, /* reset=*/U8X8_PIN_NONE);
(U8G2_R2 can be changed to suit your own screen orientation)

this works for me in the arduino IDE and with platformIO in VSCode

Fenna
